cmd/internal/flowmode/cmd_run.go

```diff
@@ -180,7 +179,13 @@ func (fr *flowRun) Run(configFile string) error {
 				return (&net.Dialer{}).DialContext(ctx, network, address)
 			}
 		},
-	})
+		Modules: module.NewModule(&module.Options{
+			Logger:    l,
+			Tracer:    t,
+			Clusterer: clusterer,
+			Reg:       reg,
+		}),
```

Comment:
I'm not a fan of this design, since it adds more complexity to the API of consuming a Flow controller. Callers shouldn't need to construct a module interface; the Flow controller should do it for them with a hidden implementation, especially since it shares the same values for the Logger/Tracer/Cluster/Registry.

Reply:
I did this to avoid circular dependencies. IMO I think I might be able to get around this/mitigate by switching to a pull based system but that seemed outside the scope of an already large PR.

Reply:
This could just be unexported code in the flow controller package, right? It doesn't need to be isolated in its own package.

Reply:
If I move the http handles back into component/module that removes the circular dependency. The other alternative is to abstract flow http but I would rather not do that at the moment.

Reply:
I bit the bullet and abstracting the flow api